### PR TITLE
Add tests for KeyToDocumentData and IndexOnlyKeyToDocumentData

### PR DESCRIPTION
**Pull request by apmoriarty**

### CODE REVIEW

**Review by avgAGB — DISMISSED** [review]

File: warehouse/query-core/src/test/java/datawave/query/function/IndexOnlyKeyToDocumentDataTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+     *            the source
+     * @param expectedSize
+     *            the expected size post-aggregation
+     */
+    private void drive(String field, Key documentKey, SortedKeyValueIterator<Key,Value> source, int expectedSize) {
+        IndexOnlyKeyToDocumentData data = new IndexOnlyKeyToDocumentData(getRange(), field, source);
+        Map.Entry<DocumentData,Document> result = data.apply(getEntry(documentKey));
+
+        int dataSize = result.getKey().getData().size();
+        assertEquals(expectedSize, dataSize);
```
Comment (ivakegg) [inline warehouse/query-core/src/test/java/datawave/query/function/IndexOnlyKeyToDocumentDataTest.java]: Should we be checking the contents of the map?

File: warehouse/query-core/src/test/java/datawave/query/function/KeyToDocumentDataTest.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+     * @param entry
+     *            the document entry
+     * @param expectedSize
+     *            the expected size post-aggregation
+     */
+    private void drive(KeyToDocumentData data, Map.Entry<Key,Document> entry, int expectedSize) {
+        Map.Entry<DocumentData,Document> result = data.apply(entry);
+
+        int dataSize = result.getKey().getData().size();
+        assertEquals(expectedSize, dataSize);
```
Comment (ivakegg) [inline warehouse/query-core/src/test/java/datawave/query/function/KeyToDocumentDataTest.java]: Content of the map check?

File: warehouse/query-core/src/test/java/datawave/query/function/IndexOnlyKeyToDocumentDataTest.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/test/java/datawave/query/function/IndexOnlyKeyToDocumentDataTest.java]: done

File: warehouse/query-core/src/test/java/datawave/query/function/KeyToDocumentDataTest.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/test/java/datawave/query/function/KeyToDocumentDataTest.java]: done

**Review by ivakegg — APPROVED** [review]

**Review by avgAGB — APPROVED** [review]

**Pull request merged**